src/serializer/ResidualHeapSerializer.js

```diff
@@ -1874,19 +1882,24 @@ export class ResidualHeapSerializer {
         );
       });
       this._objectSemaphores.set(obj, objectSemaphore);
+      emitIntegrityCommand = alternateBody => {
```

Comment:
I am completely lost here. Apart from manipulating the semaphore, what does this actually do?

Reply:
Two things:
1) It sets an alternate body.
2) It calls `.releaseOne()` on the semaphone; if the counter reaches zero, the function registered with the semaphore will run and emit the Object.freeze call into the right body.